### PR TITLE
Add a new DPCTLEvent_GetWaitList function

### PR DESCRIPTION
**Pull request by vlad-perevezentsev**
This PR adds the 'DPCTLEvent_GetWaitList' function which returns the list of events that this event waits for in the dependence graph.

### CODE REVIEW

File: dpctl-capi/include/dpctl_sycl_event_interface.h [hunk 1]
```diff
@@ -91,4 +96,15 @@ DPCTLEvent_Copy(__dpctl_keep const DPCTLSyclEventRef ERef);
 DPCTL_API
 DPCTLSyclBackendType DPCTLEvent_GetBackend(__dpctl_keep DPCTLSyclEventRef ERef);
 
+/*!
+ * @brief  Return the vector of events that this event waits for
```
Comment (oleksandr-pavlyk) [inline dpctl-capi/include/dpctl_sycl_event_interface.h:100]:
```suggestion
 * @brief  Returns the vector of events that this event still waits for
```

Comment (oleksandr-pavlyk) [reply dpctl-capi/include/dpctl_sycl_event_interface.h:100]: We should also say this is a wrapper around ``sycl::event::get_wait_list``.

File: dpctl-capi/include/dpctl_sycl_event_interface.h [hunk 2]
```diff
@@ -31,13 +31,18 @@
...
 #include "dpctl_sycl_types.h"
+#include "dpctl_vector.h"
 
 DPCTL_C_EXTERN_C_BEGIN
 
 /**
  * @defgroup EventInterface Event class C wrapper
  */
 
+// Declares a set of types abd functions to deal with vectors of
```
Comment (diptorupd) [inline dpctl-capi/include/dpctl_sycl_event_interface.h:42]: Minor typo “and” instead of “abd”

**Comment by vlad-perevezentsev** [comment]
@oleksandr-pavlyk I added a Sycl test for function.
It works on SYCL but  test fails with DPCTLSyclEvent.
```
/localdisk/work/vperevez/dp_dparray/dpctl_vlad/dpctl/dpctl-capi/tests/test_sycl_event_interface.cpp:158: Failure
Value of: DPCTLEventVector_Size(EVRef) > 0
  Actual: false
Expected: true
```

Dependencies passed via ' sycl:: handler::depends_on` are not included in the graph, but are tracked (Compiler LLVM team).
This is a bug. https://jira.devtools.intel.com/browse/CMPLRLLVM-30013

**Review by oleksandr-pavlyk — CHANGES_REQUESTED** [review]

**Review by oleksandr-pavlyk — APPROVED** [review]
I have applied the requested changes and resolved conflicts from earlier merges

**Pull request merged**